src/framework/handlers/loader.js

```diff
@@ -155,10 +184,25 @@ class ResourceLoader {
             };
 
             const normalizedUrl = url.split('?')[0];
-            if (this._app.enableBundles && this._app.bundles.hasUrl(normalizedUrl)) {
-                if (!this._app.bundles.canLoadUrl(normalizedUrl)) {
-                    handleLoad(`Bundle for ${url} not loaded yet`);
-                    return;
+            if (this._app.enableBundles && this._app.bundles.hasUrl(normalizedUrl) && !(options && options.bundlesIgnore)) {
+                // if there is no loaded bundle with asset, then start loading a bundle
+                if (!this._app.bundles.urlIsLoadedOrLoading(normalizedUrl)) {
+                    const bundles = this._app.bundles.listBundlesForAsset(asset);
+                    let bundle;
+
+                    if (options && options.bundlesFilter) {
+                        bundle = options.bundlesFilter(bundles);
+                    }
+
+                    if (!bundle) {
+                        // prioritize smallest bundle
+                        bundles?.sort((a, b) => {
+                            return a.file.size - b.file.size;
+                        });
+                        bundle = bundles[0];
```

Comment:
If bundles is `undefined` or `null` the sort will be fine from conditional chaining but indexing it at `0` will throw an error.